src/recipe/recipe.controller.ts

```diff
@@ -26,23 +28,41 @@ import {
 import { RecipeResponse } from './responses/recipe.response';
 import { User } from '@prisma/client';
 import { FindAllRecipesDTO } from './DTOs/find-all-recipes-query';
+import { FileInterceptor } from '@nestjs/platform-express';
 
 @Controller('/recipe')
 export class RecipeController {
   constructor(private readonly recipeService: RecipeService) {}
 
   @Post()
+  @UseInterceptors(FileInterceptor('file'))
   @UseGuards(AuthGuard('jwt'))
   @ApiOperation({ summary: 'Create a new recipe' })
   @ApiCreatedResponse({ type: RecipeResponse })
   @ApiUnauthorizedResponse({ description: 'Not logged in' })
   async createRecipe(
     @Body() recipe: CreateRecipeDTO,
+    @UploadedFile() file: Express.Multer.File,
     @CurrentUser() user: User,
   ): Promise<RecipeResponse> {
     return this.recipeService.createRecipe(recipe, user.id);
   }
 
+  @Post('/upload/:id')
+  @UseInterceptors(FileInterceptor('file'))
```

Comment:
Have you tested this endpoint manually with postman?
Hit use:
multipart/form-data as body

Reply:
yes i did